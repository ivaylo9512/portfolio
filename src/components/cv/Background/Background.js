import X from "../icons/X.js";
import './Background.css';
import Circles from "./Circles/Circles.js";
import SvgFilters from "./SvgFilters.js";

const Background = () => {
    return(
        <div className="background">
            <SvgFilters />
            <Circles />
            <div className="letters">
                <X/>
                <X/>
                <X/>
            </div>
        </div>
    )
}

export default Background