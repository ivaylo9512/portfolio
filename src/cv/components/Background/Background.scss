.background{
    background: url(red-black1.png) no-repeat center top;
    background-size: 75%;
    width: 100%;
    height: 100vh;
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-51.9%);
}
.background svg{
    position: absolute;
}
.letters svg:nth-of-type(1){
    width: 2.7%;
    left: 46.75%;
    top: 75.3%;
}
.letters svg:nth-of-type(2){
    width: 1.35%;
    left: 44%;
    top: 67.6%;
}
.letters svg:nth-of-type(3){
    width: 1.35%;
    left: 45.45%;
    top: 63.1%;
}