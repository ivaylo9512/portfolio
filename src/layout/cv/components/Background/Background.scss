.cv .background{
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;

    svg{
        position: absolute;
    }
    
    .mid-circle{
        width: 20%;
        left: 45%;
        top: 37%;

        circle{
            fill: var(--clr-secondary);
            filter: url(#background-circle-shadow);
        }
    }

    .letters {
        svg:nth-of-type(1){
            width: 2.7%;
            left: 46.75%;
            top: 75.3%;
        }

        svg:nth-of-type(2){
            width: 1.35%;
            left: 44%;
            top: 67.6%;
        }

        svg:nth-of-type(3){
            width: 1.35%;
            left: 45.45%;
            top: 63.1%;
        }
    }
}