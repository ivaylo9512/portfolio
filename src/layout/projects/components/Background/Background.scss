.projects-root .background{
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    overflow: hidden;
    pointer-events: none;
    z-index: -1;

    svg{
        position: absolute;
    }

    .restaurant-blob{
        fill: var(--clr-restaurant-primary);

        path{
            filter: url(#restaurant-blob-shadow);
        }
    }
    
    .profile-blob {
        fill: var(--clr-profile-primary);

        path{
            filter: url(#profile-blob-shadow);
        }
    }

    .tasks-blob{
        fill: var(--clr-tasks-primary);

        path{
            filter: url(#tasks-blob-shadow);
        }
    }

    .chat-blob{
        fill: var(--clr-chat-primary);

        path{
            filter: url(#chat-blob-shadow);
        }
    }
    
    .extensions-blob{
        fill: var(--clr-extensions-primary);

        path{
            filter: url(#extensions-blob-shadow);
        }
    }

    @media only screen and (orientation: landscape) {
        .tasks-blob{
            height: 82%;
            left: 28vh;
            top: -45%;
        }

        .restaurant-blob {
            height: 75%;
            left: 100vw;
            transform: rotate(199deg);
            top: -57%;
        }

        .extensions-blob {
            top: 68%;
            left: 200vw;
            height: 114%;
            transform: translateX(-30%) rotate(2deg);
        }

        .profile-blob {
            height: 139%;
            top: 55%;
            left: 100vw;
            transform: translate(-60%);
        }

        .chat-blob {
            width: 153vh;
            top: -117vh;
            left: 284vw;
            transform: rotate(175deg);
        }
    }

    @media only screen and (orientation: portrait) {
        .restaurant-blob{
            width: 70%;
            top: 23%;
            right: -47%;
            transform: rotate(119deg);
        }

        .tasks-blob{
            width: 48%;
            left: 6%;
            transform: translateY(-54%);
        }

        .extensions-blob{
            top: 42%;
            transform: rotate(58deg);
            left: -25%;
        }
    }
}
