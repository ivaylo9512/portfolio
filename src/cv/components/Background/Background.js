import X from "cv/icons/X.js";
import './Background.scss';
import Circles from "cv/components/Circles/Circles.js";

const Background = () => {
    return(
        <div className="background">
            <Circles />
            <div className="letters">
                <X/>
                <X/>
                <X/>
            </div>
        </div>
    )
}

export default Background