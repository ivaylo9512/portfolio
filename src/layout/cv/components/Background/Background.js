import X from '@svgIcons/X.js';
import './Background.scss';
import Circles from '@cv/components/Circles/Circles.js';
import MidLogo from '@svgIcons/MidLogo';
import ReactSvg from '@svgIcons/ReactSvg';
import O from '@svgIcons/O';

const Background = () => {
    return(
        <div className='background'>
            <Circles />
            <MidLogo />
            <div className='react-icons'>
                <ReactSvg />
                <ReactSvg />
                <ReactSvg />
            </div>
            <div className='letters bottom'>
                <X/>
                <O />
            </div>
            <div className='letters project'>
                <div>
                    <X/>
                    <O />
                </div>
                <div>
                    <X/>
                    <O />
                </div>
            </div>

        </div>
    )
}

export default Background