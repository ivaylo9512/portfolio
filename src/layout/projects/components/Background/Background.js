import ProfileBlob from '@projects/icons/ProfileBlob'
import RestaurantBlob from '@projects/icons/RestaurantBlob'
import TasksBlob from '@projects/icons/TasksBlob'
import './Background.scss'
import ExtensionsBlob from '@projects/icons/ExtensionsBlob'
import ChatBlob from '@projects/icons/ChatBlob'

const Background = () => {
    return(
        <div className='background'>
            <RestaurantBlob />
            <ProfileBlob />
            <TasksBlob />
            <ExtensionsBlob />
            <ChatBlob />
        </div>
    )
}

export default Background