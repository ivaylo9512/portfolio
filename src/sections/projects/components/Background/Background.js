import BrownBlob from "@projects/icons/BrownBlob"
import RedBlob from "@projects/icons/RedBlob"
import GreenBlob from "@projects/icons/GreenBlob"
import "./Background.scss"

const Background = () => {
    return(
        <div className="background">
            <RedBlob />
            <BrownBlob />
            <GreenBlob />
        </div>
    )
}

export default Background