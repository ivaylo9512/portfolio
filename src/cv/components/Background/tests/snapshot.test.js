import Background from "../Background";
import { shallow } from 'enzyme';

describe('Background snapshot tests', () => {
    const createWrapper = () => shallow(
        <Background />
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();
     
        expect(wrapper).toMatchSnapshot();
    })
});