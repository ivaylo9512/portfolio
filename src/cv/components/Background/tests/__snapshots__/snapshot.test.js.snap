// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background snapshot tests should match snapshot 1`] = `ShallowWrapper {}`;
